othello/apps/games/models.py

```diff
@@ -0,0 +1,173 @@
+import os
+import uuid
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.postgres.fields import ArrayField
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+from django.db.models import Q
+from django.utils import timezone
+
+from ...moderator.moderator import Player
+
+PLAYER_CHOICES = (
+    (Player.BLACK.value, "Black"),
+    (Player.WHITE.value, "White"),
+)
+
+
+def _save_path(instance, filename):
+    return os.path.join(instance.user.short_name, f"{uuid.uuid4()}.py")
+
+
+class SubmissionSet(models.QuerySet):
+    def latest(self, user=None, **kwargs):
+        if user is not None:
+            qs = self.filter(user=user, **kwargs).order_by("-created_at")
+            return qs[0] if qs.exists() else None
+        else:
+            return self.filter(**kwargs).order_by("user", "-created_at").distinct("user")
+
+    def delete(self):
```

Comment:
This should be renamed. Let's not override `QuerySet` methods to change behavior.

Reply:
Needs to override the `QuerySet` method so that if a `Submission` is deleted, its related code file is also deleted.

Reply:
Using a `post_delete` signal does work if doing a bulk delete

Reply:
*Do* we want to delete the code file when a `Submission` is deleted?

Reply:
Yeah? Removes clutter

Reply:
For purposes of record-keeping, I think we should keep old student submissions around. @theo-o ?

Reply:
I actually changed the board and symbol structure so that older scripts will have to be manually changed to fit the current structure (read the "Help" page for more info)

Reply:
Actually, It would be nice to import previous AIs. I will add a `is_legacy` field to the `Submission` model as well as add an option to set `Strategy.is_legacy = True` which will revert to the previous structure.

Reply:
alright I pushed the `is_legacy` code. I added an `is_legacy` field to the `Submission` model and it can only be edited through the admin interface/shell. If `is_legacy` is set, `PlayerRunner` will convert the board and player to the legacy versions and check if the submitted move is within [11, 88]. `PlayerRunner` will also convert the submitted move to its regular counterpart (dictionary lookup). 

Should I write a django command to import previous submissions? If so, how should I handle creating users for students who have not already logged in to the existing server?

Reply:
That was not what I meant by "old submissions." I meant that I think we should keep all *future* submissions around, even if we delete the `Submission` object. Compatibility with the old method of representing the board is not a major concern.

@theo-o what are your thoughts?

Reply:
I think keeping them around is a good idea, since we may need to audit in the future. We can always do manual cleanup as necessary when the files are no longer needed for auditing purposes (i.e. at the start of a new school year).